src/BorrowController.sol

```diff
@@ -125,22 +120,46 @@ contract BorrowController {
     }
 
     /**
-    @notice Reduces the daily limit used, when a user repays debt
-    @dev This is necessary to prevent a DOS attack, where a user borrows the daily limit and immediately repays it again.
-    @param amount Amount repaid in the market
-    */
+     * @notice Reduces the daily limit used, when a user repays debt
+     * @dev This is necessary to prevent a DOS attack, where a user borrows the daily limit and immediately repays it again.
+     * @param amount Amount repaid in the market
+     */
     function onRepay(uint amount) public {
-        uint day = block.timestamp / 1 days;
-        if(dailyBorrows[msg.sender][day] < amount) {
-            dailyBorrows[msg.sender][day] = 0;
-        } else {
-            //Safe to use unchecked, as dailyBorow is checked to be higher than amount
-            unchecked{
-                dailyBorrows[msg.sender][day] -= amount;
-            }
+        updateDailyBorrowLimit(msg.sender);
+        unchecked{
+            uint newLimit = remainingDailyBorrowLimit[msg.sender] + amount;
+            uint dailyLimit = dailyBorrowLimit[msg.sender];
+            remainingDailyBorrowLimit[msg.sender] = newLimit < dailyLimit ? newLimit : dailyLimit;
         }
     }
 
+    /**
+     * @notice Updates the daily borrow limit for the market
+     * @param market The address of the market which limit will be updated
+     */
+    function updateDailyBorrowLimit(address market) internal {
```

Comment:
Maybe if reduce most of this function's code by reusing `availableBorrowLimit()`, we can get rid of this function entirely and put its code inline within `onRepay` and `onBorrow`. This way we can avoid double writing to `remainingDailyBorrowLimit` as you wanted.

Reply:
Yeah, definitely the way to go. Added in https://github.com/InverseFinance/FiRM/pull/96/commits/a6816168f19cdad2a5747f14745f8c5be3116eb5